content-src/components/ASRouterAdmin/ASRouterAdmin.jsx

```diff
@@ -110,36 +111,49 @@ export class ASRouterAdmin extends React.PureComponent {
   renderTableHead() {
     return (<thead>
       <tr className="message-item">
-        <td>id</td>
-        <td>enabled</td>
+        <td className="min">id</td>
+        <td className="min">enabled</td>
+        <td className="min">user pref</td>
         <td>source</td>
         <td>last updated</td>
       </tr>
     </thead>);
   }
 
   handleEnabledToggle(event) {
-    const action = {type: event.target.checked ? "ENABLE_PROVIDER" : "DISABLE_PROVIDER", data: event.target.name};
+    const action = {type: event.target.checked ? "ENABLE_PROVIDER" : "DISABLE_PROVIDER", data: event.target.dataset.provider};
+    ASRouterUtils.sendMessage(action);
+    this.setState({messageFilter: "all"});
+  }
+
+  handleUserPrefToggle(event) {
+    const action = {type: "SET_PROVIDER_USER_PREF", data: {id: event.target.dataset.provider, value: event.target.checked}};
     ASRouterUtils.sendMessage(action);
     this.setState({messageFilter: "all"});
   }
 
   renderProviders() {
     const providersConfig = this.state.providerPrefs;
     const providerInfo = this.state.providers;
+    const userPrefInfo = this.state.userPrefs;
+
     return (<table>{this.renderTableHead()}<tbody>
       {providersConfig.map((provider, i) => {
         const isTestProvider = provider.id === "snippets_local_testing";
         const info = providerInfo.find(p => p.id === provider.id) || {};
+        const isUserEnabled = provider.id in userPrefInfo ? userPrefInfo[provider.id] : true;
```

Comment:
Can we just use `userPrefInfo[provider.id]` the `in + if` syntax is a bit weird to read.